pkg/spireentry/reconciler.go

```diff
@@ -110,11 +110,21 @@ func (r *entryReconciler) reconcile(ctx context.Context) {
 		}
 
 		clusterSPIFFEID.NextStatus.Stats.NamespacesSelected += len(namespaces)
+		var ignored = false
+
 		for i := range namespaces {
-			if r.config.IgnoreNamespaces.In(namespaces[i].Name) {
-				clusterSPIFFEID.NextStatus.Stats.NamespacesIgnored++
+			for _, regex := range r.config.IgnoreNamespaces {
+				if !(regex.FindString(namespaces[i].Name) == "") {
+					clusterSPIFFEID.NextStatus.Stats.NamespacesIgnored++
+					ignored = true
+					break
+				}
+			}
+
+			if ignored {
 				continue
 			}
+
```

Comment:
I dont think you need this extra `ignored` value. You can `continue` after the first regex match, that's what was being done before.

Reply:
Since there are two for loops now, continue would move onto the next iteration of the inner loop; I added the ignored value to move onto the next iteration of the outer loop (continue) after the first regex match.

Reply:
> if i'm following it fine, once you get a single one that must be ignored all next namapesces will be ignored too, is that right? and my we use regex.MathString instead?

That's correct--once the namespace matches the regex pattern for a namespace it should ignore, it won't check if it matches the pattern for other ignored namespaces. 

We used regex.MatchString earlier (commit `e330c4d381f8899730d2dea6dcab47b9eed8c35a`), but Faisal said there might be issues with error handling there. We discussed it here: https://github.com/spiffe/spire-controller-manager/pull/170#discussion_r1235720412.

Reply:
@MarcosDY I think its better to compile the regex at boot so if there's a bad regex we can error out right away.

Reply:
@unnathik Good point. I think we can move the inner for loop to a function. I think it will be easier to read that way.

Reply:
@MarcosDY good catch. Made changes to address both comments in latest commit.